src/Results/DTO/GenerativeAiResult.php

```diff
@@ -0,0 +1,405 @@
+            }
+
+            $remoteFile = $part->getRemoteFile();
+            if ($remoteFile !== null) {
+                return $remoteFile;
+            }
+        }
+
+        throw new \RuntimeException('No file content found in first candidate');
+    }
+
+    /**
+     * Converts the first candidate to an image file.
+     *
+     * @since n.e.x.t
+     *
+     * @return FileInterface The image file.
+     * @throws \RuntimeException If no image content.
+     */
+    public function toImageFile(): FileInterface
+    {
+        $file = $this->toFile();
+
+        if (!$file->getMimeType()->isImage()) {
+            throw new \RuntimeException(
+                sprintf('File is not an image. MIME type: %s', $file->getMimeType())
+            );
+        }
+
+        return $file;
+    }
+
+    /**
+     * Converts the first candidate to an audio file.
+     *
+     * @since n.e.x.t
+     *
+     * @return FileInterface The audio file.
+     * @throws \RuntimeException If no audio content.
+     */
+    public function toAudioFile(): FileInterface
+    {
+        $file = $this->toFile();
+
+        if (!$file->getMimeType()->isAudio()) {
+            throw new \RuntimeException(
+                sprintf('File is not an audio file. MIME type: %s', $file->getMimeType())
+            );
+        }
+
+        return $file;
+    }
+
+    /**
+     * Converts the first candidate to a video file.
+     *
+     * @since n.e.x.t
+     *
+     * @return FileInterface The video file.
+     * @throws \RuntimeException If no video content.
+     */
+    public function toVideoFile(): FileInterface
+    {
+        $file = $this->toFile();
+
+        if (!$file->getMimeType()->isVideo()) {
+            throw new \RuntimeException(
+                sprintf('File is not a video file. MIME type: %s', $file->getMimeType())
+            );
+        }
+
+        return $file;
+    }
+
+    /**
+     * Converts the first candidate to a message.
+     *
+     * @since n.e.x.t
+     *
+     * @return Message The message.
+     */
+    public function toMessage(): Message
+    {
+        return $this->candidates[0]->getMessage();
+    }
+
+    /**
+     * Converts all candidates to text array.
+     *
+     * @since n.e.x.t
+     *
+     * @return string[] Array of text content.
+     */
+    public function toTexts(): array
+    {
+        $texts = [];
+        foreach ($this->candidates as $candidate) {
+            $message = $candidate->getMessage();
+            foreach ($message->getParts() as $part) {
+                $text = $part->getText();
+                if ($text !== null) {
+                    $texts[] = $text;
+                    break;
+                }
+            }
+        }
+        return $texts;
+    }
+
+    /**
+     * Converts all candidates to image files.
+     *
+     * @since n.e.x.t
+     *
+     * @return FileInterface[] Array of image files.
+     */
+    public function toImageFiles(): array
```

Comment:
You added a `toFile()` above, shouldn't we also have a `toFiles()` then for parity? Arguably you could then also use that in the other methods like `toImageFiles()` and then only do the MIME type filtering in there.